app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ReviewRegistrationController.scala

```diff
@@ -50,25 +52,47 @@ class ReviewRegistrationController @Inject() (
           case Some(journeyId) =>
             for {
               incorporationDetails <- incorpIdConnector.getDetails(journeyId)
-              _                    <- updateRegistration(hasReviewedRegistration = true)
+              _                    <- markRegistrationAsReviewed()
             } yield Ok(page(request.registration, incorporationDetails))
           case _ => Future(Redirect(routes.RegistrationController.displayPage()))
         }
       else
         Future(Redirect(routes.RegistrationController.displayPage()))
     }
 
+  private def markRegistrationAsReviewed()(implicit
+    req: JourneyRequest[AnyContent]
+  ): Future[Either[ServiceError, Registration]] =
+    update { registration =>
+      val updatedMetaData =
+        registration.metaData.copy(registrationReviewed = true)
+      registration.copy(metaData = updatedMetaData)
+    }
+
+  /*
+  TODO: This will need to be refactored once:
+  1. The PPT EIS/IF Stub has been implemented
+  2. The EIS/IF and ETMP 'Create Registration/Subscription' request/response schemas are made available to us
+  3. If a submission has been successful we will need to remove the registration/submission document.
+  4. Once we have a lot more clarity. KISS.
+   */
   def submit(): Action[AnyContent] =
-    (authenticate andThen journeyAction) { _ =>
-      Redirect(routes.RegistrationController.displayPage())
+    (authenticate andThen journeyAction).async { implicit request =>
+      val referenceId = s"PPT12345678${Random.nextInt(1000000)}"
```

Comment:
Do you care about the total length of this ref id?   Do you need to pad with zeros?

Reply:
No not really. 
This will hopefully change in the near future once we get the proper ID from ETMP.
I have generated an `UUID` but the panel was designed with a smaller `String` in mind.

We will have to wait until downstream services provide us the format.